search-service/src/main/kotlin/com/egm/stellio/search/authorization/permission/web/PermissionHandler.kt

```diff
@@ -326,16 +318,26 @@ class PermissionHandler(
                     Unit.right()
             }
 
-    private suspend fun checkIsAdmin(permission: Permission): Either<APIException, Unit> =
-        permissionService.checkHasPermissionOnEntity(permission.target.id, Action.ADMIN)
-            .flatMap {
-                if (!it)
-                    AccessDeniedException(
-                        unauthorizedCreateMessage(permission.target.id)
-                    ).left()
-                else
-                    Unit.right()
+    private suspend fun checkCanModify(
+        permission: Permission,
+        isDelete: Boolean = false
+    ): Either<APIException, Unit> = either {
+        if (permission.action == Action.OWN) {
+            CHANGE_OWNER_EXCEPTION.left().bind<APIException>()
+        }
```

Comment:
This check is also sometimes done in the handler function (for instance, in `update`). And indeed it is more a business check than an access right one and should not really be here.